model/src/main/kotlin/OrtResult.kt

```diff
@@ -247,6 +247,12 @@ data class OrtResult(
             ?: getPackage(id)?.pkg?.declaredLicensesProcessed?.allLicenses?.toSortedSet()
             ?: sortedSetOf()
 
+    /**
+     * Return the detected licenses for the given package [id]. If [id] is not found an empty set is returned.
+     */
+    fun getDetectedLicensesForId(id: Identifier): SortedSet<SpdxExpression> =
```

Comment:
Why is this function needed?

Reply:
Yeah, the rationale should be part of the commit message 😉

Reply:
For easier access to the detected licenses from a `rules.kts`.

Reply:
A while ago there were many different ways the (detected) licenses from an OrtResult were accessed and used.
These many different ways call trees were then refactored to use one way of accessing the licenses. I believe this is `createLicenseInfoResolver` in `OrtResultExtensions`. 

Using the licenses via that API it makes the code independent of applied curations and excludes. 
E.g. IIUC using a `LicenseRule` instance from within `rules.kts` does access the licenses via `LicenseView` and thus via the mentioned `LicenseInfoResolver`.

Have you considered that? (I'm not against introducing that new function per se, but was afraid we're moving back towards multiple ways to access the licenses again)

Reply:
> For easier access to the detected licenses from the `rules.kts`.

It's not only that, it's also for symmetry with the existing `getConcludedLicensesForId()` and `getDeclaredLicensesForId()`. It seems odd to have helper functions for these, but not for `getDetectedLicensesForId()`.

I'd also be fine with not having `getDetectedLicensesForId()`, but then we should remove `getConcludedLicensesForId()` and `getDeclaredLicensesForId()` IMO.

Reply:
Maybe then it boils down to whether @MarcelBochtler  really needs it for the `rules.kts` or whether we do want to expose that function as API to the users. If there's no need I'd have the slight tendency to remove all these functions - however no strong opinion here, I'm fine with both.

Reply:
> Maybe then it boils down to whether @MarcelBochtler really needs it for the `rules.kts` or whether we do want to expose that function as API to the users.

I'd say it's both, bearing in mind that authors of rules also are just API users. Initially Marcel did have this code inlined in a `rules.kts`, and I was proposing to move it here for symmetry with the existing API.

Reply:
I'm strongly for removing `getDeclaredLicensesForId()` and `getConcludedLicensesForId()`, I think I just forgot to remove them after finishing the refactoring to the `LicenseInfoResolver`.
@MarcelBochtler You can already get the detected licenses with `resolvedLicenseInfo.filter(LicenseView.ONLY_DETECTED)`, this takes package configurations and settings from .ort.yml into account. If you want the raw detected licenses you could have a look at `resolvedLicenseInfo.licenseInfo.detectedLicenseInfo`.

Reply:
> I'm strongly for removing `getDeclaredLicensesForId()` and `getConcludedLicensesForId()`, I think I just forgot to remove them after finishing the refactoring to the `LicenseInfoResolver`.

Can you then please create a PR do to that removal @mnonnenmacher? I propose to close this PR then, instead of changing its code and "inverting" its scope.

Reply:
I'm ok with closing this PR if `getDeclaredLicensesForId()` and `getConcludedLicensesForId()` are also removed to keep it consistent.
I'm going to remove those functions and create another PR.